### ISSUE TITLE
Setting custom labels on created resources from command argument or env var

### ISSUE THREAD

**Issue by seboudry** [joyrex2001/kubedock#50]
Hi!

We want to configure some labels on resources created by kubedock with the current CI job information.

And we don't have control on container name or labels to used already present features.

We run kubedock as a service on Gitlab runner.
This mean that when Gitlab launch a job a pod is created with kubedock as a sidecar.
So each kubedock instance is dedicated to one job.

It would be a convenient use to configure the desired labels as environment variables or command argument of the kubedock instance.

**Comment by andreezy777** [comment]
@joyrex2001 Hi.
As I understand, to pass label, for example `com.joyrex2001.kubedock.request-memory` to kafka container we should do it through the code using `withLabel` method in `testcontainers`.
So, we need to update all our services which takes a lot of time.

If I'm correct above, then +1 for this topic, it would be great to be able to pass labels through env vars.

**Comment by joyrex2001** [comment]
@andreezy777 if you want to set kubedock specific controls, like `com.joyrex2001.kubedock.request-memory`, these usually already have an environment variable alternative available. Unfortunately not (that well) documented. In you example it would be `K8S_REQUEST_MEMORY` (see [here](https://github.com/joyrex2001/kubedock/blob/975a4588961e684f0458e5211d21ecff268c0c14/cmd/server.go#L84)).

**Comment by andreezy777** [comment]
> @andreezy777 if you want to set kubedock specific controls, like `com.joyrex2001.kubedock.request-memory`, these usually already have an environment variable alternative available. Unfortunately not (that well) documented. In you example it would be `K8S_REQUEST_MEMORY` (see [here](https://github.com/joyrex2001/kubedock/blob/975a4588961e684f0458e5211d21ecff268c0c14/cmd/server.go#L84)).

As I read in readme there are some differences between env `kubedock.request-memory` and label `com.joyrex2001.kubedock.request-memory`. Because first one - it’s about global limits and second one about limits specially for needed container.

Pls, confirm that “global” means that every container that will started by kubedock will have limits that was set up by env var , but if I need, for example apply limits X for postgres container and limits Y for Kafka container, then I need to use label through code, right?

**Comment by joyrex2001** [comment]
@andreezy777 correct, the one you would set when starting kubedock (via env, or cli argument) will be applied to all created containers. If a container adds a label itself, that will take precedence over this default.

**Comment by joyrex2001** [comment]
In https://github.com/joyrex2001/kubedock/commit/7d823eebe422d0a7f17e2e6f48b6c048966d1e64 I added support for adding labels and annotations that need to be propagated to the kubernetes pods. This can be done via cli `--label` and `--annotation` as well via adding environment variables that start with `K8S_LABEL_` and `K8S_ANNOTATION_`.

**Comment by seboudry** [comment]
Awesome :star_struck: 

When do you plan a new release?
